### PR TITLE
Read docker tag from file in root, not the previously set environment variable

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

Changes required to reflect the changes made in https://github.com/pi-hole/docker-pi-hole/pull/1212

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: advanced/Scripts/updatecheck.sh [hunk 1]
```diff
@@ -37,6 +37,8 @@ rm -f "/etc/pihole/localversions"
 VERSION_FILE="/etc/pihole/versions"
 touch "${VERSION_FILE}"
 chmod 644 "${VERSION_FILE}"
+# if /pihole.docker.tag file exists, we will use it's value later in this script
+DOCKER_TAG=$(cat file 2>/dev/null)
```
Comment (dschaper) [inline advanced/Scripts/updatecheck.sh:41]: Is there anything in the function that will prevent unexpected data that is read in from that file?

Comment (PromoFaux) [reply advanced/Scripts/updatecheck.sh:41]: I guess no more than there would have been if anyone had passed in unexpected data to `PIHOLE_DOCKER_TAG` env var. I guess expected values are `nightly`, `dev`, or version number in the format `YYYY.MM[.n]`, so we could test for that, but I wouldn't want to rule out the tag name being something arbitrary, either (in the case of extra special containers, `beta`, `experiment`, etc)

Comment (rdwebdesign) [reply advanced/Scripts/updatecheck.sh:41]: Are you worried about code execution or just the wrong text?

Comment (dschaper) [reply advanced/Scripts/updatecheck.sh:41]: I'm worried about anything and everything when we read a file in.

Comment (rdwebdesign) [reply advanced/Scripts/updatecheck.sh:41]: This file is just replacing the environment variable. Anything you can add into this file you could add to the var. 
I think the security implications are exactly the same.

Comment (yubiuser) [reply advanced/Scripts/updatecheck.sh:41]: Good point. We should take care code execution is not happening at the web interface here
https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/update_checker.php

or https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/footer.php

if arbitrary code is injected.

Comment (rdwebdesign) [reply advanced/Scripts/updatecheck.sh:41]: I created a PR (web interface) to filter values read from `/etc/pihole/versions` file.

> ... I wouldn't want to rule out the tag name being something arbitrary, either (in the case of extra special containers, beta, experiment, etc)

My code allows: Uppercase and lowercase letters, digits, dot, colon, dash, underscore and slash. I think this way we will be able to use any tag we want.

Comment (PromoFaux) [reply advanced/Scripts/updatecheck.sh:41]: I have something for this side... HMB

Comment (PromoFaux) [reply advanced/Scripts/updatecheck.sh:42]: image

**Review by dschaper — APPROVED** [review]

**Comment by dschaper** [comment]
@PromoFaux You want to squash those last two commits together?

**Comment by PromoFaux** [comment]
Sure, done. Will merge with implied approval once tests pas

**Review by dschaper — APPROVED** [review]

**Pull request merged**